app/services/content_embed_service.rb

```diff
@@ -0,0 +1,83 @@
+class ContentEmbedService
+  include ApplicationHelper
+
+  SUPPORTED_DOCUMENT_TYPES = %w[contact].freeze
+  UUID_REGEX = /([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/
+  EMBED_REGEX = /({{embed:(#{SUPPORTED_DOCUMENT_TYPES.join('|')}):#{UUID_REGEX}}})/
+
+  class EmbeddedEdition
+    attr_reader :embed_code, :document_type, :content_id
+
+    def initialize(embed_code:, document_type:, content_id:)
+      @embed_code = embed_code
+      @document_type = document_type
+      @content_id = content_id
+    end
+
+    def edition
+      @edition ||= Edition.with_document.find_by(
+        state: "published",
+        content_store: "live",
```

Comment:
Eventually we'll want this to work for draft content too - i.e. if we're processing a draft edition, we should find the latest draft or published Editions for each embed. Looks like it will be a little bit annoying to have to pass that context through the DetailsPresenter unfortunately.

Reply:
I wondered about this, but would we want draft editions to be able to embed unpublished content? What if the edition with embedded content gets published without a published edition of the embedded content existing? Presumably this problem is already solved for existing needs

Reply:
I think that's a bit of a noodle scratcher as far as UX goes - if users embed content, what would they expect to appear? My instinct would be the latest live version, but I think we'd have to test that assumption. 

If we're looking at replicating the behaviour that Contact embedding gives us in Whitehall, then there's currently _no_ concept of "draft" editions, as everything is sent directly to live, so, for now, this replicates what we already have

Reply:
Yeah, there are some complexities here for sure. I think something like:

- If a document is in draft, it should be able to refer to other documents still in draft, and it should appear together with the latest draft editions when viewed through the draft frontend / content store. I believe this currently works for edition links, but I haven't got an example to hand.
- There should be some way for the publishing UI to hint that embedded things which are in draft either don't have published versions yet, or are newer than the live versions (so the value once published may be different, unless the draft embed is published)
- If you try to publish an edition that embeds documents which are still in draft, you should get an error, telling you that you need to publish the embedded documents before you can publish the embedding document.
- It should not be possible to unpublish / withdraw documents which are embedded in other documents (so if you really wanted to do that, you'd have to find all the usages and inline the values first)